### PR TITLE
chore: migrate runner/dom.js

### PR DESCRIPTION
**Pull request by sainthkh**
- Part 2 of #21689

### User facing changelog

N/A. It's an internal refactoring.

### Additional details

- Why was this change necessary? => Some legacy codes are left inside `runner`. This PR migrates `dom` codes. 
- What is affected by this change? => N/A

### Any implementation details to explain?

To migrate `runner/dom` to `pkg/app`, these things had to be done:

  * Migrate runner/dom.js
  * Mgirate Highlight React component to Vue. 
  * Migrate AssertionsMenu React component to Vue. 
  * Remove JQuery as much as possible. 

The result is like this:

* `src/runner/dom.ts` for common codes for `Highlight` and `AssertionsMenu`.
* If it's related to placing `Highlight` correctly, I moved it inside `aut-iframe`.
* I moved `Highlight` to `runner/selector-playground`.
* I moved `AssertionsMenu` to `runner/studio`. And I splitted the file to 3 component files. 

I used `floating-ui` rather than our `Tooltip` because `floating-vue` doesn't support shadow dom. It seems that it's fixed after floating-ui@1.x.

### Steps to test

* Check if Selector Playground's `Highlight` works correctly.
* Check if Studio's Assertions menu works correctly.

### How has the user experience changed?

N/A.

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by sainthkh** [comment]
I checked the `cy.origin` test in my local Ubuntu 20.04. It hangs forever in the current `develop` branch.

**Review by emilyrohrbough — COMMENTED** [review]
@sainthkh Love the changes. This is doing a bit more than just migrating runner/dom.js. Any way we can split this into three PRs? One for migrating dom.js, one for migrating selector-playground and one for migrating studio? It'd be easier to review with smaller changes.

File: packages/frontend-shared/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
   "devDependencies": {
     "@antfu/utils": "^0.3.0",
     "@cypress-design/css": "^0.5.2",
+    "@floating-ui/dom": "1.0.1",
```
Comment (emilyrohrbough) [inline packages/frontend-shared/package.json:23]: why is this added to this package vs the app package?

Comment (sainthkh) [reply packages/frontend-shared/package.json:23]: It's my mistake and I'll fix this my `Highlight` PR.

**Comment by sainthkh** [comment]
As this PR is too big, it'll be break into 3 PRs.

* showing-Highlight-related `dom.js` functions: #23792
* Highlight component: #23973
* AssertionsMenu component: #24286
